### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by husenjanobul**

### CODE REVIEW

File: week3/homework/src/actions/deleteTodo.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const readTodosFile = require('./readTodosFile');
+const saveTodos = require('./saveTodos');
+
+
+async function deleteTodo(req, res) {
+    try {
+        const todos = await readTodosFile();
+        const todo = todos.find(todo => todo.id == req.params.id);
+        if (todo == null) {
+            res.send(`There is no todo id - ${req.params.id}`);
```
Comment (gajduk) [inline week3/homework/src/actions/deleteTodo.js]: great that you caught this, but aren't you missing something? res.send does not stop the execution of a function.

File: week3/homework/src/actions/deleteTodo.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const readTodosFile = require('./readTodosFile');
+const saveTodos = require('./saveTodos');
+
+
+async function deleteTodo(req, res) {
+    try {
+        const todos = await readTodosFile();
+        const todo = todos.find(todo => todo.id == req.params.id);
```
Comment (gajduk) [inline week3/homework/src/actions/deleteTodo.js]: 👍 my sanity thanks you

File: week3/homework/src/actions/markAsDone.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+
+
+async function markAsDone(req, res) {
+    try {
+        const todos = await readTodosFile();
+        const selectTodo = todos.find(todo => todo.id == req.params.id);
+        if (selectTodo == null) {
+            res.send(`There is no todo with id - ${req.params.id}`);
+        }
```
Comment (gajduk) [inline week3/homework/src/actions/markAsDone.js]: deja vu!?

File: week3/homework/src/actions/markAsNotDone.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const saveTodos = require('./saveTodos');
+
+
+async function markAsNotDone(req, res) {
+    try {
+        const todos = await readTodosFile();
+        const selectTodo = todos.find(todo => todo.id == req.params.id);
+        if (selectTodo == null) {
+            res.send(`There is no todo id - ${req.params.id}`);
+        }
```
Comment (gajduk) [inline week3/homework/src/actions/markAsNotDone.js]: deja vu !?

File: week3/homework/src/actions/readTodo.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+
+async function readTodo(req, res) {
+    try {
+        const todos = await readTodosFile();
+        const todo = todos.find(todo => todo.id == req.params.id);
+        if (todo == null) {
+            res.send(`There is no todo with id - ${req.params.id}`);
+        }else{
```
Comment (gajduk) [inline week3/homework/src/actions/readTodo.js]: deja vu!?

File: week3/homework/src/actions/markAsDone.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+const saveTodos = require('./saveTodos');
+
+
+
+async function markAsDone(req, res) {
+    try {
+        const todos = await readTodosFile();
+        const selectTodo = todos.find(todo => todo.id == req.params.id);
+        if (selectTodo == null) {
+            res.send(`There is no todo with id - ${req.params.id}`);
```
Comment (gajduk) [inline week3/homework/src/actions/markAsDone.js]: what is the status that we use when something is not found ?
image

File: week3/homework/src/actions/deleteTodo.js (hunk 1 above)
Comment (husenjanobul) [reply week3/homework/src/actions/deleteTodo.js]: Thank you a lot for reviewing and feedback, 
I’ll check again:)

**Pull request closed without merging**